FatPkg/EnhancedFatDxe/DiskCache.c

```diff
@@ -8,6 +8,209 @@ SPDX-License-Identifier: BSD-2-Clause-Patent
 
 #include "Fat.h"
 
+/**
+  Helper function to check if any LBA is dirty in this cache line.
+
+  @param[in]    CacheTag   - CacheTag to check
+
+  @retval       TRUE       - Cache is Dirty
+                FALSE      - Cache is not Dirty
+
+**/
+STATIC
+BOOLEAN
+IsCacheTagDirty (
+  IN CACHE_TAG  *CacheTag
+  )
+{
+  UINTN  Index;
+
+  for (Index = 0; Index < DIRTY_BLOCKS_SIZE; Index++) {

```

Comment:
Instead of a loop here, it would be more efficient to keep the BOOLEAN Dirty field that is set to FALSE when the cache line is cleared, and set to TRUE whenever SetCacheTagDirty() is called.  Then this API can return the Dirty field value.

Reply:
Sure, I can update this API and keep the Dirty field in the CACHE_TAG structure